morebits.js

```diff
@@ -3298,6 +3267,63 @@ Morebits.wiki.page = function(pageName, currentAction) {
 
 	};
 
+	// Common checks for action methods
+	// Used for move, undelete, delete, protect, stabilize
+	var fnPreflightChecks = function(action, onFailure) {
+		// if a non-admin tries to do this, don't bother
+		if (!Morebits.userIsSysop && action !== 'move') {
+			ctx.statusElement.error('Cannot ' + action + 'page : only admins can do that');
+			onFailure(this);
+			return false;
+		}
+
+		if (!ctx.editSummary) {
+			ctx.statusElement.error('Internal error: ' + action + ' reason not set (use setEditSummary function)!');
+			onFailure(this);
+			return false;
+		}
+		return true; // all OK
+	};
+
+	// Common checks for fnProcess functions (fnProcessDelete, fnProcessMove, etc.)
+	// Used for move, undelete, delete, protect, stabilize
+	var fnProcessChecks = function(action, onFailure, xml) {
+		var missing = $(xml).find('page').attr('missing') === '';
+		if ((action === 'undelete' && !missing) ||
+			(action === 'protect' && (
+				(!missing && ctx.protectCreate) ||
+					(missing && (ctx.protectEdit || ctx.protectMove)))) ||
+			missing) {
+			ctx.statusElement.error('Cannot ' + action + ' the page because it ' + (missing ? 'no longer' : 'already') + ' exists');
+			onFailure(this);
+			return false;
+		}
```

Comment:
This can be structured better by using named booleans. 
```js
var deletingNonexistingPage = action === 'delete' && missing;
var saltingExistingPage = ...
var protectingNonexisitingPage = ...
if (deletingNonexistingPage || saltingExistingPage || protectingNonexisitingPage ) {
    ....
}
```
```